### PR TITLE
[PUB-2993] Replace features.isFreeUser() by org features

### PR DESCRIPTION
**Pull request by anafilipadealmeida**
## Description
Replace features.isFreeUser() and features.isProUser() by orgs features

## Context & Notes
https://buffer.atlassian.net/browse/PUB-2993

## Checklist

-   [x] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have tested different user stories. 
-   [x] I have identified similar or related features and tested they are still working.
-   [x] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.
-   [ ] I kept accessibility in mind by following the [a11y checklist.](https://www.notion.so/buffer/Workflow-Checklist-e64d86eb795140bcbfdc16d1c72e573f)
-   [ ] I have considered [security, abuse & compliance](https://www.notion.so/buffer/Engineering-Wiki-f34142d290304c35bebadf76cc9cc89e#cc6dcc7617184227b77da2e1b262a563) implications of my changes.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/thirdParty/middleware.js [hunk 1]
```diff
@@ -68,12 +68,9 @@ export default ({ dispatch, getState }) => next => action => {
...
           ),
         });
-        const { id } = action.result;
-        const {
-          productFeatures: { planName },
-        } = getState();
+        const { id, planBase } = action.result;
         FullStory.identify(id, {
-          pricingPlan_str: planName,
+          pricingPlan_str: planBase,
```
Comment (albennett) [inline packages/thirdParty/middleware.js:73]: Curious why we decided to go with `planBase` vs `planName`. Have we always called it `planBase` in the api?

**Review by albennett — APPROVED** [review]
This looks great @anafilipadealmeida 🚀 Is the goal to get rid of the product-features package?

File: packages/thirdParty/middleware.js (hunk 1 above)
Comment (anafilipadealmeida) [reply packages/thirdParty/middleware.js:73]: That's a great question @albennett ! Some time ago the core team changed some plan related info in the api. They added a `billing_plan_tier` and a `billing_plan_base`. 
The first is what we call planName or plan, as it contains the name of each plan (eg 'premium_solo').
The second one is what we call planBase, as it is related to the group the plan is part of (free, pro or business).
Every time I don't want the particular name of the plan, but only to know to which type it belongs to, I'm using planBase 🙂 
Currently our Frontend stores the plan values, but forces pro8 and pro15 to be just 'pro'. There's already a task to fix that, to make sure we are consistent: [https://buffer.atlassian.net/browse/PUB-2990](https://buffer.atlassian.net/browse/PUB-2990)

**Comment by anafilipadealmeida** [comment]
> This looks great @anafilipadealmeida 🚀 Is the goal to get rid of the product-features package?
> 
Yes @albennett! And also to move features and plans logic as far away as possible from the frontend 😅

**Review by hamstu — APPROVED** [review]
LGTM! 💯💯

**Pull request merged**